sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarOutputWriter.scala

```diff
@@ -70,21 +72,34 @@ abstract class ColumnarOutputWriterFactory extends Serializable {
 abstract class ColumnarOutputWriter(context: TaskAttemptContext,
     dataSchema: StructType,
     rangeName: String,
-    includeRetry: Boolean) extends HostBufferConsumer {
+    includeRetry: Boolean,
+    enableAsyncWrite: Boolean = false,
+    holdGpuBetweenBatches: Boolean = false) extends HostBufferConsumer with Logging {
 
   protected val tableWriter: TableWriter
 
   protected val conf: Configuration = context.getConfiguration
 
-  // This is implemented as a method to make it easier to subclass
-  // ColumnarOutputWriter in the tests, and override this behavior.
-  protected def getOutputStream: FSDataOutputStream = {
+  private val trafficController: TrafficController = TrafficController.getInstance
```

Comment:
should this be a lazy val, that way it's not initialized at all when `enableAsyncWrite` is set to false?

Reply:
Sure. It is super cheap to create a `TrafficController` and keep it in memory, but that would be better readability-wise.